src/util/file_log.rs

```diff
@@ -97,49 +97,27 @@ impl RotatingFileLoggerCore {
         self.rollover_time = compute_rollover_time(now);
     }
 
-    fn close(&mut self) {
-        self.file.flush().unwrap()
-    }
-}
-
-/// A log implemetation which writes to file and rotates by day.
-pub struct RotatingFileLogger {
-    core: Mutex<RotatingFileLoggerCore>,
-}
-
-impl RotatingFileLogger {
-    pub fn new(file_path: &str) -> io::Result<RotatingFileLogger> {
-        let core = RotatingFileLoggerCore::new(file_path)?;
-        let ret = RotatingFileLogger {
-            core: Mutex::new(core),
-        };
-        Ok(ret)
+    fn close(&mut self) -> io::Result<()> {
+        self.file.flush()
     }
 }
 
 impl Write for RotatingFileLogger {
-    fn write(&mut self, bytes: &[u8]) -> Result<usize, ::std::io::Error> {
-        let mut core = self.core.lock().unwrap();
-        if core.should_rollover() {
-            core.do_rollover()
-        };
-        core.file.write(bytes)
+    fn write(&mut self, bytes: &[u8]) -> io::Result<usize> {
+        self.file.write(bytes)
     }
 
-    fn flush(&mut self) -> Result<(), ::std::io::Error> {
-        let mut core = self.core.lock().unwrap();
-        core.file.flush()?;
-        if core.should_rollover() {
-            core.do_rollover();
+    fn flush(&mut self) -> io::Result<()> {
```

Comment:
when do we call the flush? slog will call it directly?

Reply:
Yes https://docs.rs/slog-term/2.0.4/src/slog_term/lib.rs.html#384